### ISSUE TITLE
Bug: onError acts differently to onerror in Vanilla HTML

### ISSUE THREAD

**Issue by cat-michael-gwynne** [facebook/react#25961]
onError acts differently on <picture>, <source>, <img> in React than in vanilla HTML.

React version: 18.2.0
Chrome version: 108

## Steps To Reproduce

1. Create an HTML <picture> with a <source> with a bad URL. Add an onerror function on the <picture>, <source> and <backup <img> tags. The <img> onerror will fire.
2. Create a React <picture> with a <source> with a bad URL. Add an onError function on the <picture>, <source> and <backup <img> tags. The <img> AND <picture> onErrors will fire. 

Link to code example:

Vanilla Example (check console): https://codesandbox.io/s/adoring-villani-c88u7m?file=/index.html
React Example (check console): https://codesandbox.io/s/suspicious-glitter-bfojd6?file=/src/App.js

## The current behavior
React triggers onError for both the <picture> and <img> elements.

## The expected behavior
React should trigger onError on only the <img> element. Although, [from the looks of the spec](https://codesandbox.io/s/adoring-villani-c88u7m?file=/index.html), it should only trigger on the final <source> element.

**Comment by sunderls** [comment]
The native event doesn't bubble.
But the Synthetic Event bubbles react, this is explicitly covered in the test spec here https://github.com/facebook/react/blob/main/packages/react-dom/src/__tests__/ReactDOMEventPropagation-test.js#L777-L937

So this is not a bug in React.

but I don't have an answer why React does this, I guess this is for better consistency? It might be better to make most of the events bubble rather than have many events as non-bubble(which is the native HTML spec).

You can always use `stopPropagation()` on the synthetic event to handle it, so it should not harm.

**Comment by cat-michael-gwynne** [comment]
If this is desired, is it worth documenting?

**Comment by sunderls** [comment]
Agreed, maybe you can create a PR to update the doc here https://reactjs.org/docs/events.html

**Comment by eps1lon** [comment]
I'm getting no errors in the vanilla example. Mostly asking if `source` actually does call `onError` with just HTML.

Could you check if the behavior is the same in React 17?

**Comment by cat-michael-gwynne** [comment]
Strange, the vanilla example seems to work intermittently.
Usually when pressing enter in the demo URL bar rather than clicking refresh.

**Comment by shreya-as** [comment]
dont know****

**Comment by shreya-as** [comment]
> Agreed, maybe you can create a PR to update the doc here https://reactjs.org/docs/events.html



> I'm getting no errors in the vanilla example. Mostly asking if `source` actually does call `onError` with just HTML.
> 
> Could you check if the behavior is the same in React 17?

m

**Comment by shreya-as** [comment]
> > Agreed, maybe you can create a PR to update the doc here https://reactjs.org/docs/events.html
> 
> > I'm getting no errors in the vanilla example. Mostly asking if `source` actually does call `onError` with just HTML.
> > Could you check if the behavior is the same in React 17?
> 
>

**Comment by aralroca** [comment]
Related to https://github.com/facebook/react/issues/15446
